### ISSUE TITLE
[EKS] [fargate]: Support graviton architecture for fargate nodes spawned by EKS

### ISSUE THREAD

**Issue by mfeldheim** [aws/containers-roadmap#1629]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?

**Which service(s) is this request for?**
This could be Fargate, ECS, EKS, ECR

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.

**Are you currently working around this issue?**
How are you currently solving this problem?

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by radek-yp** [comment]
any news on this? was it added to the roadmap?

**Comment by jbg** [comment]
Got 90% through the way of configuring EKS Fargate today and then realised there's no arm64 support. Didn't even consider that this would be left out! Is there any timeline for this? We dropped x86_64 builds for some services a while ago to simplify CI.

**Comment by soloradish** [comment]
> Got 90% through the way of configuring EKS Fargate today and then realised there's no arm64 support. Didn't even consider that this would be left out! Is there any timeline for this? We dropped x86_64 builds for some services a while ago to simplify CI.

I'm in similer situation. Walked through almost all of CI pipeline settings, but found out there is no support on EKS cluster.

**Comment by radek-yp** [comment]
> Got 90% through the way of configuring EKS Fargate today and then realised there's no arm64 support. Didn't even consider that this would be left out! Is there any timeline for this? We dropped x86_64 builds for some services a while ago to simplify CI.

btw I did the same thing. Went all the way to converting my images to arm64 arch only to find out it's not possible to run them with fargate on EKS :(

**Comment by subos2008** [comment]
If they can do this and somehow drop the cost for the control plane I could finally move all my personal projects off DigitalOcean and into AWS. (Which I use for all my professional work)

**Comment by mhamann** [comment]
Yes, the organization I'm working with would really like to have this feature. We can run everything ARM, and such could take advantage of the lower-cost compute. Bummer that it isn't available. :-(

**Comment by booleanbetrayal** [comment]
We are also 100% ARM compatible, are awaiting this support, and are evaluating switching off Fargate if it isn't supported soon.

**Comment by 44int** [comment]
I was disappointed to learn that arm is not available in Fargate profile.
My development machine is also Arm-based, and I'd like to have it all. In addition, it would be a waste to have CI build both x86 and arm.

**Comment by gilbahat** [comment]
having access to this could significantly assist in our journey of embracing graviton:
currently our CI/CD system is using managed runners which is extremely convenient, but arm64 runners are not available.
with EKS-fargate-arm64 we could achieve similar fidelity by moving our container builds to EKS using Kaniko from the existing CI workflow without having to maintain and scale runners

**Comment by aiell0** [comment]
Commenting that I would also like to see this feature as I got 90% of the way to configuring running pods just to see that ARM isn't supported :(

**Comment by iancward** [comment]
coredns also recently fixed builds for arm and today cut the first release that built arm images.

- https://github.com/coredns/coredns/issues/5842#issuecomment-1375738388
- https://github.com/coredns/coredns/releases/tag/v1.10.1

Considering coredns is a long-running cluster service, switching it over to graviton/arm would make for easy cost savings.

**Comment by jacksonporter** [comment]
> Commenting that I would also like to see this feature as I got 90% of the way to configuring running pods just to see that ARM isn't supported :(

Same here, would love to see this, as ECS already has it!

**Comment by perriea** [comment]
Any news ?

**Comment by dzilbermanvmw** [comment]
++1 Would love to be able to burst into Fargate ARM fabric with EKS for short term compute intensive tasks.

**Comment by Dashing-Nelson** [comment]
+1

**Comment by visit1985** [comment]
Please stop this +1 comments…

**Comment by diwap** [comment]
+1

**Comment by diwap** [comment]
I would also like to see this feature soon as we are moving towards graviton

**Comment by eugenejen** [comment]
My company would like to see this implemented also.

**Comment by chepati** [comment]
+1

**Comment by prattcmp** [comment]
Amazon is really slacking here.

**Comment by imhyeons** [comment]
A lot of companies in Korea are hesitant to switch arch to FARGATE. just because it doenst support ARM architectures

**Comment by booleanbetrayal** [comment]
Just to chime in here, we abandoned Fargate for Managed Node Groups and ARM64 was the straw that broke that camel's back.

**Comment by alanprot** [comment]
+!

**Comment by dc2tom** [comment]
Disappointed for this one not to appear at re:invent - it just seems so odd for it to have existed happily for ECS for so long to still be missing from EKS.

**Comment by Deepak13245** [comment]
+1

**Comment by diwap** [comment]
me too

**Comment by visit1985** [comment]
@tabern, can you maybe add this to the containers-roadmap project?

**Comment by igor9silva** [comment]
Even though we are not able to explicit request an ARM node, is it possible to achieve it?

For example, using an ARM-only container imagem?

**Comment by nantiferov** [comment]
We would be glad to see this feature implemented too. Majority of our clusters are Graviton-only and Fargate nodes only kept on amd64 due to this.

**Comment by mfeldheim** [comment]
> Just to chime in here, we abandoned Fargate for Managed Node Groups and ARM64 was the straw that broke that camel's back.

We decided to do the same. Managed node groups have other advantages besides that, we cut out cost in comparison to fargate and pods spin up much faster

**Comment by KristopherBorja** [comment]
My company would immediately onboard and use this feature. +1

**Comment by cb-sanjaikumar** [comment]
We really need this feature since most of the org's are moving towards graviton for the cost and perf benefits. Bringing EKS Fargate with graviton support will really help us to reap the benefits that we are already get in other graviton platforms.
